src/main/java/net/datafaker/idnumbers/BulgarianIdNumber.java

```diff
@@ -0,0 +1,62 @@
+package net.datafaker.idnumbers;
+
+import net.datafaker.providers.base.BaseProviders;
+
+import java.time.LocalDate;
+
+/**
+ * <a href="https://en.wikipedia.org/wiki/Unique_citizenship_number">Specification</a>
+ */
+public class BulgarianIdNumber implements IdNumbers {
+    private static final int[] CHECKSUM_WEIGHTS = {2, 4, 8, 5, 10, 9, 7, 3, 6};
+    private static final int[] EVEN_DIGITS = {0, 2, 4, 6, 8};
+    private static final int[] ODD_DIGITS = {1, 3, 5, 7, 9};
+
+    public String getValid(BaseProviders faker) {
+        String basePart = basePart(faker);
+        return basePart + checksum(basePart);
+    }
+
+    public String getInvalid(BaseProviders faker) {
+        String basePart = basePart(faker);
+        return basePart + (checksum(basePart) + 1) % 10;
+    }
+
+    private String basePart(BaseProviders faker) {
+        LocalDate birthDate = faker.date().birthdayLocalDate(0, 200);
```

Comment:
Would it be an option to use the new TimeAndDate provider here? Also, why can someone's age be 199 or 200?